fix(curriculum): clarify instructions and update tests for Step 27 (#62240)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-registration-form/60fab8367d35de04e5cb7929.md

# --description--

> Add an anchor `a` with the text `Read our terms and conditions` before the newly added `label`. Link the text `terms and conditions` to:

Reviewer: ```suggestion
Add an anchor element with the text `Read our terms and conditions` before the newly added `label`. Set the `href` to:
```

the whole text is inside the anchor element, so the instructions need to be adjusted, as you can't link only part of the text

```md
https://www.freecodecamp.org/news/terms-of-service/
```

Then add the text `I accept the terms and conditions` immediately after the input element in the label.

# --hints--

> You should use an `a` element to link to the terms and conditions with the text `Read our terms and conditions` before the `label` that follows the third `fieldset`.

Reviewer: ```suggestion
You should have an `a` element right before the `label` with the text `Read our terms and conditions`.
```

```js

> assert.exists(document.querySelector('fieldset:nth-child(3) + a')?.innerText.trim(), 'Read our terms and conditions');

Reviewer: ```suggestion
assert.match(document.querySelector('fieldset:nth-child(3) + a')?.innerText.trim().replaceAll(/\s+/g, ' '), /Read our terms and conditions\.?/i);
```

let's make things easier for everyone, and make the dot at the end optional, and don't care about multiple spaces.

```

The `a` element should have an `href` of `https://www.freecodecamp.org/news/terms-of-service/`.

```js
assert.match(document.querySelector('fieldset:nth-child(3) + a')?.href, /https:\/\/www\.freecodecamp\.org\/news\/terms-of-service\/?/);
```

> You should have an `input` element inside the `label` that follows the `a` tag.

Reviewer: ```suggestion
You should have an `input` element inside the `label`.
```

```js
assert.exists(document.querySelector('fieldset:nth-child(3) + a + label > input'));
```

> Then, inside the same `label`, add the text `I accept the terms and conditions` immediately after the input element.

Reviewer: hints are not instructions

```suggestion
Inside the `label`, immediately after the `input` element you should have the text `I accept the terms and conditions`.
```

```js

> assert.include(document.querySelector('fieldset:nth-child(3) + a + label')?.textContent.trim(), 'I accept the terms and conditions');

Reviewer: ```suggestion
assert.include(document.querySelector('fieldset:nth-child(3) + a + label')?.textContent.trim().replaceAll(/\s+/g, ' '), 'I accept the terms and conditions');
```

```

The text inside your anchor element has extra leading or trailing whitespace. The only spaces in the anchor text should be between the words `terms`, `and`, and `conditions`.

```js
const nestedAnchor = document.querySelector('fieldset:nth-child(3) + a')?.textContent;
const innerContent = nestedAnchor?.innerHTML;
assert.isNotTrue(/^\s+|\s+$/.test(innerContent));

> ```

Reviewer: let's remove this test, as already said earlier, please

> ```

Reviewer: if you decide to ignore a comment, please comment on why

> ```

Reviewer: that is not correct, you can see in the Files tab that the test is still present in the file
